data-raw/update-compdata.R

```diff
@@ -6,14 +6,63 @@ library(gcamdata.compdata)
 # in COMPDATA and won't be deleted automatically. We have a good reason for
 # doing it this way, though.
 
-data("COMPDATA")   # load as an object that we'll modify and write back out below
+COMPDATA <- get_all_compdata()   # load as an object that we'll modify and write back out below
 
 outputs_dir <- "./outputs"
 comment_char <- "#"
 for(output_fn in list.files(outputs_dir, pattern = "*.csv", full.names = TRUE)) {
     dataname <- sub('.csv$', '', basename(output_fn))
     data <- read_csv(output_fn, comment = comment_char)
-    COMPDATA[[dataname]] <- data
+    # we will store the individual data compressed in the list but not compress
+    # the .rda file at the end so that when it gets loaded back into memory for
+    # tests it will still be compressed and not be a massive memory hog that hangs
+    # around
+    # users will use gcamdata.compdata::get_comparison_data to retrieve the data
+    # which will handle decompression
+    conn <- rawConnection(raw(0), "w")
+    saveRDS(data, file = conn)
+    data_compress <- memCompress(rawConnectionValue(conn), "xz")
+    close(conn)
+    COMPDATA[[dataname]] <- data_compress
 }
 
-use_data(COMPDATA, overwrite = TRUE, internal = FALSE, compress = "xz")
+# The maximum bin size in bytes we want to allow, we need to restrict this as we will
+# need to commit this data to git
+MAX_BIN_SIZE <- 100 * 1024^2
+
+# calculate the minimum number of bins required to stay under the max
+total_size <- sum(sapply(COMPDATA, object.size))
+NUM_BINS <- ceiling(total_size / MAX_BIN_SIZE)
+
+# partition data into bins
+compdata_bins <- list()
+for(i in seq_len(NUM_BINS)) {
+  compdata_bins[[i]] <- list()
+  compdata_bins[[i]][["size"]] <- 0
+  compdata_bins[[i]][["data"]] <- list()
+}
+
+# TODO: a more sophisticated approach to ensure bins are evenly balanced
+curr_bin <- 1
+for(n in names(COMPDATA)) {
+  curr_size <- object.size(COMPDATA[[n]])
+  if(compdata_bins[[curr_bin]][["size"]] + curr_size > (total_size / NUM_BINS) && curr_bin < NUM_BINS) {
+    curr_bin <- curr_bin + 1
```

Comment:
Should we have a sanity check that the object size does not exceed the max bin size, all by itself?